tldw_Server_API/app/core/StudyPacks/source_resolver.py

```
@@ -0,0 +1,213 @@
+                    label=label,
+                    evidence_text=_pick_evidence_text(selection, "\n\n".join(evidence_parts)),
+                    locator={
+                        "media_id": media_id,
+                        "chunk_id": chunk_id,
+                        "chunk_index": chunk_index,
+                    },
+                )
+
+        timestamp_value = locator.get("timestamp_seconds")
+        if timestamp_value is None:
+            raise ValueError("Media source requires either chunk locators or timestamp_seconds")
```

Ruff (0.15.7)

[warning] Avoid specifying long messages outside the exception class

(TRY003)